lib/client/renderer.js

```diff
@@ -490,29 +575,121 @@ function init (client, d3) {
+                  console.log(result); 
+                  chart().drag.selectAll('.arrow').transition().duration(5000).style('opacity', 0).remove();
+                }
+              );
+            } else {
+              chart().drag.selectAll('.arrow').remove();
             }
-          );
-        } else {
-          chart().drag.selectAll('.arrow').remove();
+            break;
+          case "Remove":
+            if (window.confirm(translate('Remove treatment ?'))) {
+              client.socket.emit(
+                'dbRemove', 
+                { 
+                  collection: 'treatments', 
+                  _id: treatment._id
+                },
+                function callback(result) { 
+                  console.log(result); 
+                  chart().drag.selectAll('.arrow').transition().duration(5000).style('opacity', 0).remove();
+                }
+              );
+            } else {
+              chart().drag.selectAll('.arrow').remove();
+            }
+            break;
+          case "Move insulin":
+            if (window.confirm(translate('Change insulin time to %1 ?', { params: [newTime.toLocaleTimeString()] } ))) {
+              if (treatment.carbs) {
+                dboperation = 'dbUpdateUnset';
+              } else {
+                dboperation = 'dbRemove';
+              }
+              client.socket.emit(
+                dboperation, 
+                { 
+                  collection: 'treatments', 
+                  _id: treatment._id,
+                  data: { insulin: 1 } 
+                }
+              );
+              var newTreatment = _.cloneDeep(treatment);
+              delete newTreatment._id;
+              delete newTreatment.carbs;
+              newTreatment.created_at = newTime.toISOString();
+              client.socket.emit(
+                'dbAdd', 
+                { 
+                  collection: 'treatments', 
+                  data: newTreatment 
+                },
+                function callback(result) { 
+                  console.log(result); 
+                  chart().drag.selectAll('.arrow').transition().duration(5000).style('opacity', 0).remove();
+                }
+              );
+            } else {
+              chart().drag.selectAll('.arrow').remove();
+            }
+            break;
+          case "Move carbs":
+            if (window.confirm(translate('Change carbs time to %1 ?', { params: [newTime.toLocaleTimeString()] } ))) {
+              if (treatment.insulin) {
+                dboperation = 'dbUpdateUnset';
+              } else {
+                dboperation = 'dbRemove';
+              }
+              client.socket.emit(
+                dboperation, 
+                { 
+                  collection: 'treatments', 
+                  _id: treatment._id,
+                  data: { carbs: 1 } 
+                }
+              );
+              var newTreatment = _.cloneDeep(treatment);
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Variable 'newTreatment' is already declared in line 640.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1849529815/issues/source?bid=2452379&fileBranchId=3084282#l674)